app/models/course/group.rb

```diff
@@ -0,0 +1,18 @@
+class Course::Group < ActiveRecord::Base
+  stampable
+
+  before_validation :set_defaults, if: :new_record?
```

Comment:
`after_initialize` and `before_validate`, and allow it to run twice. Sometimes not all the arguments are given at initialise.